src/test/librbd/test_mirroring.cc

```diff
@@ -644,3 +644,50 @@ TEST_F(TestMirroring, MirrorStatusList) {
   ASSERT_EQ(0, m_rbd.mirror_image_status_list(m_ioctx, last_read, 4096, &images));
   ASSERT_EQ(0U, images.size());
 }
+
+TEST_F(TestMirroring, RemoveBootstrapped)
+{
+  ASSERT_EQ(0, m_rbd.mirror_mode_set(m_ioctx, RBD_MIRROR_MODE_POOL));
+
+  uint64_t features = RBD_FEATURE_EXCLUSIVE_LOCK | RBD_FEATURE_JOURNALING;
+  int order = 20;
+  ASSERT_EQ(0, m_rbd.create2(m_ioctx, image_name.c_str(), 4096, features,
+                             &order));
+  librbd::ImageCtx *ictx =
+      new librbd::ImageCtx(image_name, "", nullptr, m_ioctx, false);
+  ASSERT_EQ(0, ictx->state->open());
+  ASSERT_EQ(-EBUSY, m_rbd.remove(m_ioctx, image_name.c_str()));
+
+  // simulate the image is open by rbd-mirror bootstrap
+  uint64_t handle;
+  struct MirrorWatcher : public librados::WatchCtx2 {
+    MirrorWatcher(librbd::ImageCtx *ictx) : m_ictx(ictx) {
+    }
+    virtual void handle_notify(uint64_t notify_id, uint64_t cookie,
+                               uint64_t notifier_id, bufferlist& bl) {
+      // received IMAGE_UPDATED notification from remove
+      if (!m_closed) {
+	FunctionContext *ctx = new FunctionContext(
+	  [this, notify_id, cookie](int r) {
+	    EXPECT_EQ(0, r);
+	    bufferlist bl;
+	    m_ictx->md_ctx.notify_ack(RBD_MIRRORING, notify_id, cookie, bl);
+	  });
+	m_closed = true;
+	m_ictx->state->close(ctx);
```

Comment:
Probably overkill since rbd-mirror will never be able to close the image before acking the notification.